packages/components/inputs/select-input/src/select-input.tsx

```diff
@@ -403,7 +409,13 @@ const SelectInput = (props: TSelectInputProps) => {
               ...props.components,
             } as ReactSelectProps['components']
           }
-          menuIsOpen={props.isReadOnly ? false : undefined}
+          menuIsOpen={
+            props?.menuIsOpen && !props.isReadOnly
+              ? props.menuIsOpen
+              : props.isReadOnly
+              ? false
+              : undefined
+          }
```

Comment:
NIT: Here's probably a cleaner version:
```suggestion
          menuIsOpen={
            props.isReadOnly ? false : !!props.menuIsOpen
          }
```

Reply:
Looks way cleaner, will change that

Reply:
I tried that and it breaks the behaviour as the undefined state is missing

Reply:
Actually, something like below should also work. 

`props.menuIsOpen && !props.isReadOnly`

Reply:
Just wanted to mention that boolean casting is not needed:
```jsx
menuIsOpen={props.isReadOnly ? false : props.menuIsOpen}
```

Reply:
yeah. it's not. the second condition is even better. 

`props.menuIsOpen && !props.isReadOnly`